### ISSUE TITLE
Authorization callout user not valid (when having users in nats options)

### ISSUE THREAD

**Issue by hspaay** [nats-io/nats-server#4479]
### What version were you using?

Latest from main as of 2023-09-02.

### What environment was the server running in?

dev box Ubuntu

### Is this defect reproducible?

Yes. Take a look at auth.go:293
```go
for _, u := range opts.AuthCallout.AuthUsers {
			// Check for user in users and nkeys since this is server config.
			var found bool
			if len(s.users) > 0 {
				_, found = s.users[u]
			} else if len(s.nkeys) > 0 && !found {
				_, found = s.nkeys[u]
			}
			if !found {
				s.Errorf("Authorization callout user %q not valid: %v", u, err)
			}
		}
```
The statement:
> else if len(s.nkeys) > 0 && !found {

is never run when len(s.users) > 0.

This should simply be an "if len()..." without the else.

### Given the capability you are leveraging, describe your expectation?

No errors should be reported if the callout user exists in nkeys

### Given the expectation, what is the defect you are observing?

I'm observing an incorrect error in the log. The callout nkey does exist, but the check doesn't check the nkeys.

**Comment by tbeets** [comment]
I am not able to reproduce this issue. AuthCallout responder identified by NKEY succeeds in below server configuration having both nkey and user defined.   Can you clarify your test configuration @hspaay ?
```
		listen: "127.0.0.1:-1"
		server_name: A
		authorization {
			timeout: 1s
			users: [	{ nkey: "UATYCZZNVF4FZK73AQZTCRNESSNYH5IP6TLQOGEJQMP2BJFI5OYIJMBY" }, 
						{ user: "scott", password: "tiger" }]
			auth_callout {
				# Needs to be a public account nkey, will work for both server config and operator mode.
				issuer: "ABJHLOVMPA4CI6R5KLNGOB4GSLNIY7IOUPAJC4YFNDLQVIOBYQGUWVLA"
				# users that will power the auth callout service.
				auth_users: [ "UATYCZZNVF4FZK73AQZTCRNESSNYH5IP6TLQOGEJQMP2BJFI5OYIJMBY" ]
			}
		}
```

**Comment by hspaay** [comment]
@tbeets Thank you for looking into this.

Your  understanding is impeccable :) My setup uses the embedded server without config file where predefined users and nkeys are added programmatically and callout is used for dynamically added users/nkeys/etc.

On options reload, the validation is tripped and indicates a problem where there is none (as far as my understanding goes). Thankfully it is not fatal otherwise it would be a  show stopper. However the logs contain errors that aren't and might confuse the reading.

**Comment by tbeets** [comment]
Thanks @hspaay

Confirmed that errant ERR message printed with reload signal but not at initial start.

```
$ nats-server --config ./temp.conf
[33252] 2023/09/05 12:35:02.471782 [INF] Starting nats-server
[33252] 2023/09/05 12:35:02.471814 [INF]   Version:  2.10.0-beta.49
[33252] 2023/09/05 12:35:02.471816 [INF]   Git:      [not set]
[33252] 2023/09/05 12:35:02.471818 [INF]   Name:     A
[33252] 2023/09/05 12:35:02.471820 [INF]   ID:       NDHNMR5OZWHPNDKRU57QOLGQJX3SJGHAT4X5MRHKEPGFVNAI3OZLZN35
[33252] 2023/09/05 12:35:02.471827 [WRN] Plaintext passwords detected, use nkeys or bcrypt
[33252] 2023/09/05 12:35:02.471834 [INF] Using configuration file: ./temp.conf
[33252] 2023/09/05 12:35:02.472115 [INF] Listening for client connections on 127.0.0.1:4222
[33252] 2023/09/05 12:35:02.472126 [INF] Server is ready
[33252] 2023/09/05 12:35:56.740073 [INF] Reloaded: authorization users
[33252] 2023/09/05 12:35:56.740085 [INF] Reloaded: accounts
[33252] 2023/09/05 12:35:56.740114 [ERR] Authorization callout user "UATYCZZNVF4FZK73AQZTCRNESSNYH5IP6TLQOGEJQMP2BJFI5OYIJMBY" not valid: <nil>
[33252] 2023/09/05 12:35:56.740502 [INF] Reloaded server configuration
```

**Comment by tbeets** [comment]
@aricart @derekcollison Will create a PR for the validation logic fix for now which will clear spurious ERR on reload of this issue.

**Comment by tbeets** [comment]
@hspaay The validation logic fix will be in 2.10.0.  Thank you for the report.
